server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -861,6 +862,7 @@ protected void abandonSegment(final long truncatedTime, final Sink sink)
         );
         for (FireHydrant hydrant : sink) {
           cache.close(SinkQuerySegmentWalker.makeHydrantCacheIdentifier(hydrant));
+          hydrant.getSegment().close();
```

Comment:
`abandonSegment()` is called in `FlushingPlumber` not from `mergeExecutor`. Well, _probably_ it doesn't lead to race, because in the context of `FlushingPlumber` `mergeExecutor` is not used at all, but it's a dangerous situation. Could you please refactor `RealtimePlumber`/`FlushingPlumber` by extracting logic and fields which are used by both into a superclass, and make `RealtimePlumber` and `FlushingPlumber` both subclasses of that class. So that `FlushingPlumber` doesn't have unused `Executor` fields.